onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -76,7 +73,14 @@ class TensorrtExecutionProvider : public IExecutionProvider {
 
   std::shared_ptr<KernelRegistry> GetKernelRegistry() const override;
 
+  void SetMaxBatchSize(const int max_batch_size) {
+      max_batch_size_ = max_batch_size;
+  }
+
  private:
+ int max_batch_size_ = 1;
+ const int max_workspace_size_ = 1 << 30;
```

Comment:
wouldn't we want to make workspace_size configurable too?

Reply:
Per TensorRT doc, max workspace size_ should be set as large as possible and there is no need to make it configurable. Actually memory is allocated as needed when creating an IExecutionContext.

Reply:
1 << 30 doesn't strike me to mean "as large as possible" 
how did you choose that value? if it can't guarantee it will work for all cases, then it would be safer to make it configurable.

Reply:
actually we should be considering the reverse scenario.
on devices that don't have as much memory (e.g. jetson nano) , one may want to limit the size of the workspace to be small.
it's better to make this configurable to cover both sides of the spectrum.